sql/core/src/main/scala/org/apache/spark/sql/sources/ddl.scala

```diff
@@ -303,6 +303,9 @@ private[sql] object ResolvedDataSource {
       mode: SaveMode,
       options: Map[String, String],
       data: DataFrame): ResolvedDataSource = {
+    if (data.schema.map(_.dataType).exists(_.isInstanceOf[IntervalType])) {
+      sys.error("cannot save dataframe with interval type out to external systems")
```

Comment:
this is shared by sql as well, so we shouldn't use dataframe in the error message. Maybe

"Cannot save interval data type into external storage."